security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisImpl.java

```diff
@@ -69,70 +97,120 @@ public boolean removeInterceptor(SecurityAnalysisInterceptor interceptor) {
+                                    setPreContigencyOkAndCheckViolations(resultBuilder);
+                                }, computationManager.getExecutor())
+                                .thenComposeAsync(
+                                    aVoid -> submitAllLoadFlows(workingStateId, contingenciesProvider, postContParameters, resultBuilder),
+                                    SCHEDULER_EXECUTOR);
+                    } else {
+                        return setPreContingencyKo(resultBuilder);
+                    }
+                }).thenApply(aVoid -> resultBuilder.build());
+    }
 
-                        resultBuilder.preContingency()
-                                .setComputationOk(true);
-                        violationDetector.checkAll(network, resultBuilder::addViolation);
-                        resultBuilder.endPreContingency();
-
-                        List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
-
-                        futures = new CompletableFuture[contingencies.size()];
-
-                        String hash = UUID.randomUUID().toString();
-                        for (int i = 0; i < contingencies.size(); i++) {
-                            Contingency contingency = contingencies.get(i);
+    private void setPreContigencyOkAndCheckViolations(SecurityAnalysisResultBuilder resultBuilder) {
+        resultBuilder.preContingency().setComputationOk(true);
+        violationDetector.checkAll(network, resultBuilder::addViolation);
+        resultBuilder.endPreContingency();
+    }
 
-                            String postContStateId = hash + "_" + contingency.getId();
+    private CompletableFuture<Void> setPreContingencyKo(SecurityAnalysisResultBuilder resultBuilder) {
+        resultBuilder.preContingency().setComputationOk(false).endPreContingency();
+        return CompletableFuture.completedFuture(null);
+    }
 
-                            // run one loadflow per contingency
-                            futures[i] = CompletableFuture
-                                    .supplyAsync(() -> {
-                                        network.getVariantManager().cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, postContStateId);
-                                        network.getVariantManager().setWorkingVariant(postContStateId);
+    private CompletableFuture<Void> submitAllLoadFlows(String workingStateId,
+            ContingenciesProvider contingenciesProvider,
+            LoadFlowParameters postContParameters,
+            SecurityAnalysisResultBuilder resultBuilder) {
+        List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
+        List<String> variantIds = makeWorkingVariantsNames(contingencies.size());
+        BlockingQueue<String> queue = new ArrayBlockingQueue<>(variantIds.size(), false, variantIds);
+        network.getVariantManager().cloneVariant(workingStateId, variantIds);
```

Comment:
You should not create a variant for all variant, but create 1 variant per worker. This is a regression on what you did yesterday?